Composer/packages/extension/src/extensionRegistration.ts

```diff
@@ -58,12 +58,17 @@ export class ExtensionRegistration {
    * Publish related features
    *************************************************************************************/
   public async addPublishMethod(plugin: PublishPlugin) {
-    log('registering publish method', this.name);
-    this.context.extensions.publish[plugin.customName || this.name] = {
+    if (this.context.extensions.publish[plugin.name]) {
+      throw new Error(`Duplicate publish method. Cannot register publish method with name ${plugin.name}.`);
```

Comment:
nit: if this is surfaced to the UI, maybe use formatMessage

Reply:
Couple of things:

- It doesn't get to the UI. It will end up in the terminal if this is thrown.
- Our server is not yet set up to handle i18n.

Reply:
As far as I've seen, messages that get thrown out to the console (like those inside exceptions) don't need to be localized. Only strings that get displayed to a normal user do.